File: LeetCode/Easy/1_two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}

        for i, n in enumerate(nums):
            req_num = target - n
            if req_num in visited:
                return [visited[req_num], i]
            else:
                visited[n] = i
        

File: LeetCode/Easy/1_two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const visited = new Map();

    for(let i = 0; i < nums.length; i++) {
        const req_num = target - nums[i];
        
        if (visited.has(req_num)) {
            return [i, visited.get(req_num)];
        }

        visited.set(nums[i], i);
    }

    return [];
};

// console.log(twoSum([2, 7, 11, 15], 18));
